### PR TITLE
signing with nonce

### PR DESCRIPTION
**Pull request by saiakilesh**
Signing `hash(nonce, publicKey)`

### CODE REVIEW

File: contracts/utils/Governable.sol [hunk 1]
```diff
@@ -63,30 +64,40 @@ contract Governable {
...
     }
 
     /**
      * @dev Transfers ownership of the contract to a new account (`newOwner`).
      * Can only be called by the current owner.
      */
-    function transferOwnershipWithSignature(address newOwner, bytes memory sig) public {
+    function transferOwnershipWithSignature(address newOwner, uint32 nonce, bytes memory sig) public {
+        require(refreshNonce < nonce, "Invalid nonce");
+        require(nonce <= refreshNonce + 1, "Nonce must increment by 1");
```
Comment (drewstone) [inline contracts/utils/Governable.sol:80]: We need to add checks that nonce doesn't increase tremendously (causing overflow attack).

**Review by drewstone — CHANGES_REQUESTED** [review]
Can we make sure to hound more nonce updates for all proposals necessary including this one?

File: contracts/utils/Governable.sol (hunk 1 above)
Comment (drewstone) [reply contracts/utils/Governable.sol:80]: Oops I had it backwards. This looks good.

**Comment by saiakilesh** [comment]
> Can we make sure to hound more nonce updates for all proposals necessary including this one?

What does hound more nonce updates mean? I didn't understand.

**Comment by drewstone** [comment]
Bound*

**Comment by saiakilesh** [comment]
> Bound*

Ah ok cool.

**Review by drewstone — APPROVED** [review]

**Pull request merged**